src/CMakeLists.txt

```diff
@@ -38,4 +41,6 @@ if (USE_DLIB)
   list(APPEND ddetect_SOURCES backends/dlib/DNNStructures.h backends/dlib/dliblib.cc backends/dlib/dliblib.h backends/dlib/dlibmodel.cc backends/dlib/dlibmodel.h backends/dlib/dlibinputconns.h)
 endif()
 add_library(ddetect ${ddetect_SOURCES})
-add_dependencies(ddetect caffe_dd)
+if (USE_CAFFE)
+  add_dependencies(ddetect caffe_dd)
+endif()
```

Comment:
An error happen when compiling with CAFFE2 and not CAFFE:
pytorch is set as a dependencie of caffe_dd (to ensure they share the same protobuf files)
But if CAFFE is not built, pytorch still need to be compiled.

Adding a
```add_dependencies(ddetect pytorch)```
should solve this